src/rest_vol_link.c

```diff
@@ -609,10 +608,17 @@ RV_link_get(void *obj, const H5VL_loc_params_t *loc_params, H5VL_link_get_args_t
             by_idx_data.is_recursive               = FALSE;
             by_idx_data.index_type                 = loc_params->loc_data.loc_by_idx.idx_type;
             by_idx_data.iter_order                 = loc_params->loc_data.loc_by_idx.order;
-            by_idx_data.idx_p                      = &loc_params->loc_data.loc_by_idx.n;
             by_idx_data.iter_function.link_iter_op = NULL;
             by_idx_data.op_data                    = NULL;
 
+            /* Avoid compiler warnings about const pointer reassignment */
+            hsize_t idx_p;
+
+            if (0 > memcpy(&idx_p, &loc_params->loc_data.loc_by_idx.n, sizeof(hsize_t)))
```

Comment:
No need to check anything here, just do the `memcpy`. `memcpy` returns a pointer to `&idx_p` so checking for `< 0` here isn't correct. We likely don't need to do any `memcpy` here to fix this warning though.